documentation/release_notes.markdown

```diff
@@ -4,116 +4,104 @@ title: "Puppet Server: Release Notes"
+-   [EZ-48: Add systemd services for debian 8 and ubuntu >= 15.04](https://tickets.puppetlabs.com/browse/EZ-48)
 
-* [SERVER-1150](https://tickets.puppet.com/browse/SERVER-1150)
+### Improvements: Bootstrap interfaces and behaviors
 
-### All changes
-
-* [All Puppet Server issues targeted at this release](https://tickets.puppet.com/issues/?jql=project%20%3D%20SERVER%20AND%20fixVersion%20%3D%20%22SERVER%202.3.2%22%20ORDER%20BY%20updated%20DESC%2C%20priority%20DESC%2C%20created%20ASC)
-
-## Puppet Server 2.3.1
-
-Released March 21, 2016.
+Trapperkeeper 1.4.0 introduces improvements to its [service bootstrapping][] feature, which Puppet Server uses to manage its certificate authority (CA) service. These improvements include better error tolerance, improved logging of bootstrap-related errors, and the ability to read `bootstrap.cfg` settings from multiple configuration files and directories.
 
-This is a bug-fix release that resolves a disruptive logging configuration issue.
+For more information, see the following [Trapperkeeper project](https://tickets.puppetlabs.com/browse/TK) tickets:
 
-### Bug fix: Puppet Server starts when configured to log to syslog
+-   [TK-211: Trapperkeeper doesn't error if two services implementing the same protocol are started](https://tickets.puppetlabs.com/browse/TK-211)
+-   [TK-347: Support directories and paths in TK's "bootstrap-config" CLI argument](https://tickets.puppetlabs.com/browse/TK-347)
+-   [TK-349: TK should not fail during startup if an unrecognized service is found in bootstrap config](https://tickets.puppetlabs.com/browse/TK-349)
+-   [TK-351: Ensure all bootstrap related errors log what file they come from](https://tickets.puppetlabs.com/browse/TK-351)
 
-If its Logback service is configured to log to syslog, Puppet Server 2.3.0 fails to start. Puppet Server 2.3.1 fixes this regression, which did not affect prior versions of Puppet Server.
-
-* [SERVER-1215](https://tickets.puppet.com/browse/SERVER-1215)
-
-### All changes
+Also, see these related tickets:
 
-* [All Puppet Server issues targeted at this release](https://tickets.puppet.com/issues/?jql=project%20%3D%20SERVER%20AND%20fixVersion%20%3D%20%22SERVER%202.3.1%22%20ORDER%20BY%20updated%20DESC%2C%20priority%20DESC%2C%20created%20ASC)
+-   [EZ-72: Implement support for overriding bootstrap-config path in app's project.clj](https://tickets.puppetlabs.com/browse/EZ-72)
 
-## Puppet Server 2.3.0
+### Improvement: Responses to unauthenticated HTTPS requests include less information
 
-Released March 16, 2016.
+When responding to unauthorized HTTPS requests, previous versions of Puppet Server 2.x returned the requester's IP address and [authorization rule][auth.conf] in addition to logging the failed request. Puppet Server 2.4.0 removes this information from the response and directs the responder to consult the server logs for details.
```

Comment:
It might be worth noting that this is *only* if they have `use-legacy-auth-conf: false` in puppetserver.conf. In the future it will be true whether or not they're using that, once PUP-6150 is done.